### DISCUSSION TITLE
Error Loading MET on derecho

### DISCUSSION THREAD

**Discussion by CPKalb** [dtcenter/METplus#2497] [Existing Builds] [answered]
Hi:

I have been trying to load MET on derecho using the commands on the existing builds page:
module use /glade/work/dtcrt/METplus/derecho/components/MET/installations/modulefiles
module load met/11.1.0

However, when I execute these commands, I get an error:
Lmod has detected the following error:  These module(s) or extension(s) exist but cannot be loaded as requested:
"cray-mpich/8.1.27"
   Try: "module spider cray-mpich/8.1.27" to see how to load the module(s).

Executing this command requires loading "cray-mpich/8.1.27" which failed while processing the following module(s):

    Module fullname         Module Filename
    ---------------         ---------------
    intel-classic/2023.0.0  /glade/u/apps/derecho/modules/23.06/Core/intel-classic/2023.0.0.lua
    met/11.1.0              /glade/work/dtcrt/METplus/derecho/components/MET/installations/modulefiles/met/11.1.0

Are there additional commands needed to get MET to work on derecho?

Thanks,
Tina

**Comment by jprestop** [comment, marked as answer]
Hi @CPKalb.  Thanks for posting this Discussion.  With the completion of the upgrade on Derecho, CISL announced:

> During the outage, new user software has been installed including new versions of the Intel, NVIDIA, and Cray compilers. The new software is available in version 23.09 of the software stack, indicated by the top-level ncarenv/23.09 module. This ncarenv/23.09 version is now the default module environment on Derecho, and is thus loaded at login time. Older versions of software packages are still available, but you must load the ncarenv/23.06 module first to have access to them. This change means that if you have not been explicitly loading the module for your desired ncarenv version in your job scripts, you may get errors suggesting that certain modules are not available, or you may see new versions of packages (e.g., intel, netcdf) loaded unexpectedly. As a reminder, CISL recommends always loading specific versions of all necessary modules at the start of a job script; doing so helps document the job environment and increases reproducibility.

I plan to reinstall MET-11.1.0 on Derecho to utilize `ncarenv/23.09` instead of the current `ncarenv/23.06` which it utilizes now.  You may be able to load `ncarenv/23.06` first, before loading MET, but I have not tried it.    I hope to reinstall on Derecho sometime soon and will update here once the reinstallation is complete.

**Reply by jprestop** [reply]
@CPKalb I see that in a Slack message you mentioned that loading the following before loading MET worked for you:
```
module load ncarenv/23.06
module reset
```
It's good to know that this temporary work around works in the meantime.

**Reply by jprestop** [reply]
Hi @CPKalb. I have reinstalled MET-11.1.0 on Derecho and it should be working as described on the [Existing Builds](https://dtcenter.org/community-code/metplus/metplus-5-1-existing-builds) page now.  Please note that this version of MET is actually built using the Intel Classic compilers, and not the Intel oneApi compilers as I stated earlier in a meeting. This is because our compilation script for MET-11.1.0 doesn't have support for oneApi.  When I install the MET-12.0.0 betas and the official release, those will use the default Intel oneApi compilers.   Please also note that the METdataio, METcalcpy, and METplotpy as also now installed on Derecho as well.
